admin/app.py

```diff
@@ -12,388 +12,372 @@
+from admin import utils
 
 load_dotenv()
 
-app = Flask(__name__)
-app.config['SECRET_KEY'] = os.getenv('SECRET_KEY')
-app.config['PERMANENT_SESSION_LIFETIME'] = timedelta(minutes=90)
-app.config['MAX_CONTENT_LENGTH'] = 10 * 1024 * 1024  # 10 Mb limit
 
-connect(
-    db=os.getenv('DB_NAME'),
-    host=os.getenv('MONGO_URL'),
-    port=int(os.getenv('PORT'))
-)
-
-login = LoginManager(app)
-login.login_view = 'admin_login'
-login.init_app(app)
-
-ROWS_PER_PAGE = 6
-
-
-@app.errorhandler(404)
-def page_not_found(e):
-    prev_url = request.referrer
-    query = parse.parse_qs(parse.urlparse(prev_url).query)
-    page = query.get('page')
-    user_search = query.get('userSearch')
-    book_search = query.get('bookSearch')
-
-    if page and page[0].isdigit() or user_search or book_search:
-        url_to_redirect = prev_url.split('?')[0] + '?'
-        if page:
-            page_value = int(page[0]) - 1
-            url_to_redirect += f'&{"page"}={page_value}'
-        if user_search:
-            user_search_value = user_search[0]
-            url_to_redirect += f'&{"userSearch"}={user_search_value}'
-        if book_search:
-            book_search_value = book_search[0]
-            url_to_redirect += f'&{"bookSearch"}={book_search_value}'
-
-        return redirect(url_to_redirect)
-
-    return 'Page not found'
-
-
-@app.route('/')
-@login_required
-def start_page():
-    num_users = User.objects.count()
-    num_books = Book.objects.count()
-    num_active_users = User.objects(status=Status.ACTIVE).count()
-    num_inactive_users = User.objects(status=Status.INACTIVE).count()
-    num_muted_users = User.objects(status=Status.MUTED).count()
-    num_active_books = Book.objects(status=Status.ACTIVE).count()
-    num_inactive_books = Book.objects(status=Status.INACTIVE).count()
-
-    statistics = Statistics(num_users, num_books, num_active_users, num_inactive_users, num_muted_users,
-                            num_active_books, num_inactive_books)
-
-    return render_template('index.html', statistics=statistics)
-
-
-@app.route('/users_list')
-@login_required
-def get_users_list():
-    users = utils.search_and_pagination(collection=User, order_field='email')
-    return render_template('users_list.html', users=users)
-
-
-@app.route('/active_users_list')
-@login_required
-def get_active_users_list():
-    users = utils.search_and_pagination(collection=User, order_field='email', status=Status.ACTIVE)
-    return render_template('users_list.html', users=users)
-
-
-@app.route('/inactive_users_list')
-@login_required
-def get_inactive_users_list():
-    users = utils.search_and_pagination(collection=User, order_field='email', status=Status.INACTIVE)
-    return render_template('users_list.html', users=users)
-
-
-@app.route('/create_user', methods=['GET', 'POST'])
-@login_required
-def create_user():
-    form = AddUserForm(request.form)
-    if request.method == 'POST' and form.validate_on_submit():
+def create_app(db_name, url, port):
```

Comment:
Do I understand it correct, that the only change here is wrapping this app into `create_app` function ?
And it will be good to add some comments to this PR to add more clarity to this change.

Reply:
Yes. We should implement the Application Factory Pattern to have the ability to create and use the developed app in the tests. "If you move the creation of this object into a function, you can then create multiple instances of this app later." Flask Documentation.
I didn't find another way to test the flask application